mavisp/modules.py

```diff
@@ -105,63 +105,57 @@ def ingest(self, mutations):
+            all_methods = []
+            models = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method))
 
         # check that:
             # all methods are supported
             # there are no duplicated methods (possible since they are in different model dirs)
-        for model in models:
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-            if not set(method_dirs).issubset(set(self.methods.keys())):
-                this_error = f"One or more {self.name} methods are not supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-            all_methods.extend(method_dirs)
-
-        if len(all_methods) != len(set(all_methods)):
-                this_error = f"Only using one single instance of any given method is supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-
-        for model in models:
-
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-
-            for method_dir in method_dirs:
-                self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
-                self.data = self.data.join(self.methods[method_dir].data)
+            for model in models:
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                if not set(method_dirs).issubset(set(self.methods.keys())):
+                    this_error = f"One or more {self.name} methods are not supported"
+                    raise MAVISpMultipleError(warning=warnings,
+                                            critical=[MAVISpCriticalError(this_error)])
+                all_methods.extend(method_dirs)
+                if len(all_methods) != len(set(all_methods)):
+                                this_error = f"Only using one single instance of any given method is supported"
+                                raise MAVISpMultipleError(warning=warnings,
+                                                        critical=[MAVISpCriticalError(this_error)])
+            for model in models:
+
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                for method_dir in method_dirs:
+                    self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
```

Comment:
I think we need to change the design philosophy a bit here. Right now, calling `parse` in a `Method` object assigns the parsed data to `self.data` in the object. This design was chosen because we had only one instance of `Methods` object per method (e.g. one FoldX, one Rosetta...). Now we have multiple of them, meaning that each time we run `Method.parse` its `self.data` is overwritten, which might lead down the line to misrerepresnet/misunderstand what is actually stored in the method object (which of the 3 FoldX instances is in the Method object *now*?)

so it would be good if we could refactor it like this:

`Method.parse` should `return` the results of the parsing instead of assigning to `self.data`. In this way this part of code would look something like:

```
this_data = self.methods[method_dir].parse(...)
self.data = self.data.join(this_data, ...)
```

for this to work all the `Method` classes need to be refactored to return rather than assign